src/index.rs

```diff
@@ -272,9 +271,13 @@ impl IndexTable {
 		key
 	}
 
-	pub fn get(&self, key: &Key, sub_index: usize, log: &impl LogQuery) -> (Entry, usize) {
-		log::trace!(target: "parity-db", "{}: Querying {}", self.id, hex(&key));
-		let key = u64::from_be_bytes((key[0..8]).try_into().unwrap());
+	pub(crate) fn get(&self, key: &TableKey, sub_index: usize, log: &impl LogQuery) -> (Entry, usize) {
```

Comment:
This function (and others in this module) should still accept `Key` rather than `TableKey`. Index always deals with real hashed keys.